File: MedicApp/Models/AppointmentHistory.cs
using MedicApp.Enums;

namespace MedicApp.Models
{
    public class AppointmentHistory
    {
        public Guid Id { get; set; }
        public Guid? AppointmentId { get; set; }
        public bool IsStartedAppointment { get; set; }    
        public bool IsFinishedAppointment { get; set; }    
        public Guid ChangedByUser_RefID { get; set; }
        public bool IsDeleted { get; set; }
        public DateTime Creation_TimeStamp { get; set; }

        public AppointmentHistory()
        {
            Id = Guid.NewGuid();
            Creation_TimeStamp = DateTime.Now;
            IsDeleted = false;
        }
    }

    public class AppointmentListHistory
    {
        public Guid Id { get; set; }
        public Guid? AppointmentId { get; set; }
        public Guid ClinicId { get; set; }
        public DateTime TimeFinished { get; set; }
        public string ClinicName { get; set; }
        public string ReportDescription { get; set; }
    }
}

File: Client/src/app/model/AppointmentHistory.ts
export interface AppointmentHistoryInterface {
  id?: string;
  appointmentId?: string;
  clinicId?: string;
  clinicName?: string;
  reportDescription?: string;
  timeFinished?: string;
  rating?: number;
  workingHours?: string;
}

export class AppointmentHistory implements AppointmentHistoryInterface {
  id?: string;
  appointmentId?: string;
  clinicId?: string;
  clinicName?: string;
  reportDescription?: string;
  timeFinished?: string;
  rating?: number;
  workingHours?: string;

  constructor(obj: AppointmentHistoryInterface) {
    this.id = obj.id;
    this.appointmentId = obj.appointmentId;
    this.clinicId = obj.clinicId;
    this.clinicName = obj.clinicName;
    this.reportDescription = obj.reportDescription;
    this.timeFinished = obj.timeFinished;
    this.rating = obj.rating;
    this.workingHours = obj.workingHours;
  }
}
